lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -454,53 +432,44 @@ sub _get_obs_builds_text {
     return join ', ', @builds;
 }
 
-sub _concurrency_guard {
-    my $app = shift->app;
+sub _concurrency_guard ($app) {
```

Comment:
I'm wondering why that works.
Before the first argument was the controller, so it was like:
```
my ($c) = @_;
my $app = $c->app;
```
Now you are assigning the controller directly to `$app`.
Maybe that works because both the controller and app have access to the helper function? (_concurrency_guard is used as a helper).

The line where that is used is covered, but can you please check that it does the right thing?
If it is covered by tests then it might be enough to add some debugging here and run the according test.

Reply:
I think, since we didn't have proper unit tests for this specific function nor integration test that can cover all the interaction we have, this kind of think can happen and we wont get errors as long as the function in consideration is not actually used.

I've used a find and replace within a vim query and I only check after if the changes made sense. I've fixed the assignation now and I'm going to check the correct assignation during the morning.